### DISCUSSION TITLE
exécution du code de transactions monitoring

### DISCUSSION THREAD

**Discussion by TARIQ-BARKI** [frmscoe/Full-Stack-Docker-Actio#30] [General]
je n'arrive pas à distinguer les fichiers à exécuter et les étapes à suivre pour exécuter la partie du système de surveillance des transactions en temps réel

**Comment by Justus-at-Tazama** [comment]
> je n'arrive pas à distinguer les fichiers à exécuter et les étapes à suivre pour exécuter la partie du système de surveillance des transactions en temps réel

Bonjour! Malheureusement, je ne parle pas Français

If you are attempting to deploy the Docker container for a public instance of Tazama, please follow the step-by-step instructions in the README in .this repository: https://github.com/frmscoe/Full-Stack-Docker-Tazama

**Reply by TARIQ-BARKI** [reply]
Thank's for your answer. Actually, i finished all the steps but for the last one i have an error : cannot read properties of undefined (reading'report') at test-script.
Do you suggest any solution please

**Reply by Justus-at-Tazama** [reply]
Hi Tariq,

That error means that your end-to-end process in the evaluation flow was interrupted and the evaluation report was not produced in the database. (I'll update the test script at some point to make the test fail a bit more gracefully to be more clear about the error that occurs.)

## Newman may be running faster than Tazama on your machine

Are you executing the request through Newman with the `--delay-request 500` option? The evaluation of a transaction is completed only after the initial 200 OK response is received when the pacs.002 transaction is submitted. Sometimes, the test runner checks the database for the result before the evaluation is completed and the test then fails with the result you see above. If you want to check if the test result was eventually delivered, you can open the evaluationResults.transactions database.collection in your instance of Arango:

1. Go to http://localhost:18529 (the default Arango DB deployment location) on your browser
2. Click on _system in the top right to get to the database selection page
3. Select `evaluationResults`:

![image](https://github.com/user-attachments/assets/9595bab7-acf5-467b-8887-d6203bc05753)

4. Click `transactions` to open the database collection:

![image](https://github.com/user-attachments/assets/17845d60-ab07-410e-974f-8dace91655ff)

5. Click on the `Content` to browse the data:

![image](https://github.com/user-attachments/assets/b40cb77e-56f8-4c4f-b76a-a3af7ac1801c)

If there are no records listed here, your end-to-end flow was interrupted somewhere along the way. We then have to look at the container logs for each of the containers in the end-to-end flow to see if we can spot which container failed to deliver the results. The containers are, in order:

1. tms (Transaction Monitoring Service API)
2. ed (Event Director)
3. rule-901 (Rule Processor 901)
4. tp (Typology Processor)
5. tadproc (Transaction Aggregation and Decisioning Processor)

You should see a logged entry in each of the container logs to show your transaction arriving.

## Your containers may have restarted in the wrong order

What also sometimes happens is when you stop the Docker service and then restart it later, for example after rebooting your machine. Docker sometimes (often!) starts up all the services in a different order to which they were originally created. In the Tazama stack, ArangoDB, Redis and Nats *must* be started up before any of the other system services. You can try to stop all your services manually and then start them up again in the same order that they were created:

1. arango redis nats
2. tms, ed, rule-901, tp, tadp

The tp and the tadp are particularly sensitive to this dependency - they use the configuration information in the ArangoDB at startup to define the NATS routing.

## We may have broken something in our latest release

We released Tazama 2.0.0 a couple of weeks ago and while we tested the release on our side, it is possible that we may have broken something. We're checking now and I'll update you if we find anything and release an update.

**Reply by Justus-at-Tazama** [reply]
Hi @TARIQ-BARKI 

We discovered a couple of issues in our Full-Stack deployment and are in the process of fixing those.

Firstly, our ArangoDB database installation scripts are incompatible with the latest release of ArangoDB. We are unsure it this is an issue on our side or theirs and we're logging it for them to advise. In the meantime, we're explicitly specifying release 3.11.10.1 as the ArangoDB version in our Docker deployment scripts until we can address the issue in ArangoDB 3.12.

Secondly, the Postman scripts that can be used to manually update the configurations in the databases have an error in the configuration of the typology processor. Our latest breaking changes to the typology processor were not followed through to the Postman collection. We're pushing a patch now and it will go up in the next few minutes. We are also flagging the Postman setup step as optional in the repository readme in the future. If the ArangoDB setup scripts are running when you deploy ArangoDB, you don't need to run the Postman configuration as well.

**Reply by TARIQ-BARKI** [reply]
thank you 
It works for the test where all transactions have an amount of 0, but I also want to test scenarios involving fraud and alerts.
how can i add thoses types of transactions

**Comment by Justus-at-Tazama** [comment]
@TARIQ-BARKI - please also consider joining our Slack at https://slack.tazama.org - we are usually much better at responding directly to messages there. If you have any trouble with the deployment of the Docker container, please let me know.
